lib/spack/docs/tutorial_modules.rst

```diff
@@ -776,113 +1103,169 @@ After modifications the configuration file will be:
 
-  Due to MODULEPATH changes the following have been reloaded:
+  Due to MODULEPATH changes, the following have been reloaded:
     1) netlib-scalapack/2.0.2-openblas
 
-This layout is already a great improvement over the usual non-hierarchical layout,
-but it still has an asymmetry: ``LAPACK`` providers are semantically the same as ``MPI``
-providers, but they are still not part of the hierarchy.  We'll see a possible solution
-next.
+``Lmod`` took care of swapping the MPI provider for us, and it also substituted the
+``netlib-scalapack`` module to conform to the change in the MPI.
+In this way we can't accidentally pull-in two different MPI providers at the
+same time or load a module file linked to ``openmpi`` when ``mpich`` is also loaded.
+Consistency for compilers and MPI is ensured by the tool.
+
 
 .. Activate lmod and turn the previous modifications into lmod:
    Add core compilers
 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Extend the hierarchy to other virtual providers
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Add LAPACK to the hierarchy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The hierarchy just shown is already a great improvement over non-hierarchical layouts,
+but it still has an asymmetry: ``LAPACK`` providers cover the same semantic role
+as ``MPI`` providers, but yet they are not part of the hierarchy.
+
+To be more practical, this means that despite we have gained an improved consistency in
+our environment when it comes to ``MPI``, we still have the same problems as we had before
+for ``LAPACK`` implementations:
 
-.. warning::
-  This is an experimental feature
-    Having a hierarchy deeper than ``Core``/``Compiler``/``MPI`` is an experimental
-    feature, still not fully supported by ``module spider``,
-    see `here <https://github.com/TACC/Lmod/issues/114>`_. Furthermore its use
-    with hierarchies more complex than ``Core``/``Compiler``/``MPI``/``LAPACK``
-    has not been thoroughly tested in production environments.
+.. code-block:: console
+
+  root@module-file-tutorial:/# module list
+
+  Currently Loaded Modules:
+    1) gcc/7.2.0   2) openblas/0.2.20   3) openmpi/1.10.2   4) netlib-scalapack/2.0.2-openblas
 
-Spack permits you to generate Lua hierarchical module files where users
-can add an arbitrary list of virtual providers to the triplet
-``Core``/``Compiler``/``MPI``. A configuration file like:
+  root@module-file-tutorial:/# module load netlib-scalapack/2.0.2-netlib
+  Autoloading netlib-lapack/3.6.1
+
+  The following have been reloaded with a version change:
+    1) netlib-scalapack/2.0.2-openblas => netlib-scalapack/2.0.2-netlib
+
+  root@module-file-tutorial:/# module list
+
+  Currently Loaded Modules:
+    1) gcc/7.2.0   2) openblas/0.2.20   3) openmpi/1.10.2   4) netlib-lapack/3.6.1   5) netlib-scalapack/2.0.2-netlib
+
+Now: having hierarchies that are deeper than ``Core``/``Compiler``/``MPI`` is
```

Comment:
"Now: having hieararchies..." -> "Hierarchies..." and also "is probably" -> "are probably"